src/boxerror_replacement.rs

```diff
@@ -1,11 +1,14 @@
-//! Experimental replacement for `Box<dyn Error>` that implements `Error`
+//! Experimental replacement for `Box<dyn Error>` that implements `Error`.
+
 use std::error::Error;
 use std::fmt;
 
 type BoxError = Box<dyn Error + Send + Sync + 'static>;
 
+/// Wrapper type for a `BoxError`.
```

Comment:
I don't think the documentation here should focus on this being a wrapper for a `Box<dyn Error>` instead it should be focused on this being an Owned `dyn Error`.

We will also want to add a section talking about how it differs from `Box<dyn Error>`, the main distinctions being that it implements the `Error` trait and that it doesn't implement `From<E: Error>`. We should probably also go into the why, and explain why box dyn Error doesn't implement these traits, and how we use the associated `Result` type and the new `Try` trait to offer the same automatic conversions provided directly on `Box<dyn Error>` without sacrificing our ability to implement the `Error` trait.